pkg/proctree/nodes/process.go

```diff
@@ -0,0 +1,239 @@
+package nodes
+
+import "sync"
+
+// Arguments to feed or retrieve a process node
+
+type ProcessInfo struct {
+	Tid       int32
+	NsTid     int32
+	Pid       int32
+	NsPid     int32
+	StartTime uint64
+	ExitTime  uint64
+}
+
+type FileInfo struct {
+	Name      string
+	Path      string
+	Device    uint32
+	Ctime     uint64
+	Inode     uint64
+	InodeMode uint16
+}
+
+type ProcessForkData struct {
+	Parent ProcessInfo
+	Child  ProcessInfo
+}
+
+type ProcessExecData struct {
+	Command     FileInfo
+	Interpreter FileInfo
+	Pid         int32
+	StartTime   uint64
+}
+
+type ProcessExitData struct {
+	ExitedPid int32
+	ExitCode  int64
+	StartTime uint64
+	ExitTime  uint64
+	GroupExit bool
+}
+
+//
+// Process Node
+//
+
+type Process struct {
+	current     uint32         // current hash
+	parent      uint32         // parent hash
+	info        *ProcessInfo   // current task basic info
+	command     *FileInfo      // current task command info
+	interpreter *FileInfo      // current task interpreter info
```

Comment:
// TODO: will add here (in near future) information about:
- opened files and their characteristics
- opened sockets and network connections
- anything regarding an existing process
- processes and their information will be kept in memory as LRU map (the map[hash]*Process will be a LRU) allows